src/Analyzers/CSharp/Analyzers/UseTupleSwap/CSharpUseTupleSwapDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseTupleSwap
+{
+    /// <summary>
+    /// Looks for code of the form:
+    /// 
+    /// <code>
+    ///     var temp = expr_a;
+    ///     expr_a = expr_b;
+    ///     expr_b = temp;
+    /// </code>
+    ///
+    /// and converts it to:
+    /// 
+    /// <code>
+    ///     (expr_b, expr_a) = (expr_a, expr_b);
+    /// </code>
+    /// 
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpUseTupleSwapDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        public CSharpUseTupleSwapDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.UseTupleSwapDiagnosticId,
+                   EnforceOnBuildValues.UseTupleSwap,
+                   CSharpCodeStyleOptions.PreferTupleSwap,
+                   LanguageNames.CSharp,
+                   new LocalizableResourceString(
+                       nameof(CSharpAnalyzersResources.Use_tuple_to_swap_values), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(compilationContext =>
+            {
+                // Tuples are only available in C# 7 and above.
+                var compilation = compilationContext.Compilation;
+                if (((CSharpCompilation)compilation).LanguageVersion < LanguageVersion.CSharp7)
+                    return;
+
+                context.RegisterSyntaxNodeAction(
```

Comment:
Shouldn't this be `compilationContext.RegisterSyntaxNodeAction`? Or are these equivalent in this case?

Anyway I always liked @sharwell suggestion to shadow the original `context` name (ie, rename `compilationContext` to `context`)

Reply:
it's the same. but i agree that shadowing is cleaner.